commit b25da7eb39ca97f9e33d4e9ba8e1aeee9f8bd560
Author: bertin <macportsuser@macports.org>
Date:   Sun May 25 22:22:15 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-linux-no-default-compdeps.diff

diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index 25877a353..93e12fbaf 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -818,8 +818,10 @@ proc portconfigure::configure_get_default_compiler {} {
     global compiler.blacklist compiler.fallback compiler.whitelist
     if {${compiler.whitelist} ne ""} {
         set search_list ${compiler.whitelist}
+        ui_debug "Picking compiler from whitelist ${search_list}"
     } else {
         set search_list ${compiler.fallback}
+        ui_debug "Picking compiler from fallbacks ${search_list}"
     }
     foreach compiler $search_list {
         set allowed yes
@@ -1374,6 +1376,7 @@ proc portconfigure::get_compiler_fallback {} {
 
     # Check our override
     if {[info exists default_compilers]} {
+        ui_debug "Taking fallback compilers from default_compilers=${default_compilers}"
         return $default_compilers
     }
 
@@ -1384,6 +1387,7 @@ proc portconfigure::get_compiler_fallback {} {
         } else {
             set available_apple_compilers [portconfigure::get_apple_compilers_xcode_version]
         }
+        ui_debug "available_apple_compilers=${available_apple_compilers}"
         set system_compilers [list]
         foreach c ${available_apple_compilers} {
             set vmin [portconfigure::get_min_command_line $c]
@@ -1421,7 +1425,11 @@ proc portconfigure::get_compiler_fallback {} {
             }
         }
     }
-    set compilers [list]
+    if {[option os.platform] eq "darwin"} {
+        set compilers [list]
+    } else {
+        set compilers [list "cc"]
+    }
     lappend compilers {*}${system_compilers}
     # when building for PowerPC architectures, prefer GCC to Clang
     set cur_arch ${configure.build_arch}
@@ -1716,7 +1724,15 @@ proc portconfigure::add_automatic_compiler_dependencies {} {
 port::register_callback portconfigure::add_automatic_compiler_dependencies
 # and an option to turn it off if required
 options configure.compiler.add_deps
-default configure.compiler.add_deps yes
+default configure.compiler.add_deps {[portconfigure::configure_get_add_deps]}
+proc portconfigure::configure_get_add_deps {} {
+    if {[option os.platform] eq "darwin"} {
+        return yes
+    } else {
+        return no
+    }
+}
+
 # helper function to add dependencies for a given compiler
 proc portconfigure::add_compiler_port_dependencies {compiler} {
     global os.major porturl
